engine/consensus/approvals/approvals_cache.go

```diff
@@ -0,0 +1,79 @@
+package approvals
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+
+	lru "github.com/hashicorp/golang-lru"
+)
+
+// ApprovalsCache is a wrapper over `lru.Cache` that provides needed api for processing result approvals
+// Extends functionality of `lru.Cache` but mainly using grouping of calls and providing more user friendly interface.
+type ApprovalsCache struct {
+	cache *lru.Cache
+}
+
+func NewApprovalsCache(limit uint) *ApprovalsCache {
+	cache, _ := lru.New(int(limit))
+	return &ApprovalsCache{
+		cache: cache,
+	}
+}
+
+func (c *ApprovalsCache) TakeIf(approvalID flow.Identifier, predicate func(*flow.ResultApproval) bool) *flow.ResultApproval {
```

Comment:
Since it's a wrapper over lru cache, I expect `nil` if there is no result approval.
I think `Take` is not needed anymore, it's a leftover of previous impl